drivers/include/extend/gpio.h

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2018 Acutam Automation, LLC
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_extend_gpio GPIO extension
+ * @ingroup     drivers_extend
+ * @brief       GPIO peripheral extension handling
+ *
+ * The GPIO extension interface makes handling of non-CPU GPIO devices invisible
+ * to code using the GPIO periph API (periph/gpio.h). This is accomplished by
+ * reserving part of the range of values of gpio_t. When a call to the GPIO API
+ * uses a pin that falls within this range, it is parsed into a device ID that
+ * is looked up in the GPIO extension list and the call is redirected to the
+ * corresponding device.
+ *
+ * @{
+ *
+ * @file
+ * @brief       GPIO extension interface definitions
+ *
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ */
+
+#ifndef EXTEND_GPIO_H
+#define EXTEND_GPIO_H
+
+#include <limits.h>
+#include <stdint.h>
+
+#include "periph_conf.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef GPIO_EXT_DEV_MASK
+/**
+ * @brief   Set device mask in gpio_t
+ */
+#define GPIO_EXT_DEV_MASK    (gpio_t)((UINT_MAX << GPIO_EXT_DEV_LOC) \
+                                      & GPIO_EXT_THRESH)
+#endif
+
+#ifndef GPIO_EXT_PIN_MASK
+/**
+ * @brief   Set pin mask in gpio_t
+ */
+#define GPIO_EXT_PIN_MASK    (gpio_t)(~(UINT_MAX << GPIO_EXT_DEV_LOC))
+#endif
+
+/**
+ * @brief   Callback typedef for gpio_init
+ *
+ * @see @ref drivers_periph_gpio
```

Comment:
You might think about to replace
```suggestion
 * @see #gpio_init
```
to link to the function directly.

Reply:
Fixed.